### PR TITLE
[REF] web: Clean company switcher widget

### PR DESCRIPTION
**Pull request by LucasLefevre**
The company switcher was recently added for the mobile version.
It was merged in a stable release (13.0). This meant the diff was kept
to a minimum and the code quality had to accommodate with our stable policy.

This commit refactors the company switcher widget to be able to reuse
it for the mobile version.

The widgets are converted to OWL components and the "dropped menu" of
the desktop widget is reused for the mobile widget.

Task 2205603

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/widgets/switch_company_menu.js [hunk 1]
```diff
@@ -8,72 +8,164 @@ odoo.define('web.SwitchCompanyMenu', function(require) {
...
+
+        this.mainCompanyId = mainCompanyId;
+        this.mainCompanyName = mainCompanyId ? this._getCompanyName(mainCompanyId) : '';
+        return this._super(...arguments);
+    },
+
+    async start() {
+        this.component = new ComponentWrapper(this, this.componentClass, this._getComponentProps());
+        await this.component.mount(this.el);
+        // this.$el.remove();
```
Comment (LucasLefevre) [inline addons/web/static/src/js/widgets/switch_company_menu.js]: commented code

File: addons/web/static/src/js/widgets/switch_company_menu.js [hunk 2]
```diff
@@ -8,72 +8,164 @@ odoo.define('web.SwitchCompanyMenu', function(require) {
...
+    constructor() {
+        super(...arguments);
+        this.state = useState({
+            isMobile: config.device.isMobile,
+        });
+    }
+
+}
+SwitchCompanyComponent.components = { SwitchCompanyListComponent };
+SwitchCompanyComponent.pops = {
```
Comment (jvm-odoo) [inline addons/web/static/src/js/widgets/switch_company_menu.js]:
```suggestion
SwitchCompanyComponent.props = {
```
😁

Comment (LucasLefevre) [reply addons/web/static/src/js/widgets/switch_company_menu.js]: Ho well spotted, thanks!

**Comment by LucasLefevre** [comment]
Old branch

**Pull request closed without merging**